server/channels/api4/channel.go

```diff
@@ -87,6 +87,12 @@ func createChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// check if required fields are present
+	if channel.TeamId == "" || channel.Name == "" || channel.DisplayName == "" {
```

Comment:
I think we should just verify the `Type` and `TeamId` to let the permission checks pass. The rest of the validation is taken care of by the store and app layers. Let's also add a unit test please. Thanks.